src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -263,6 +261,16 @@ private <T extends SyntaxElement> T parse(Iterator<? extends SyntaxElementInfo<?
 								x = x2;
 							}
 							T t = info.c.newInstance();
+							if (getParser().getCurrentStructure() != null) {
```

Comment:
I think we should still check for any Structure restrictions the syntax element has

Reply:
Sorry for the delay in reply, I was wondering exactly what you mean by this sorry?